src/java.base/share/classes/sun/security/pkcs12/PKCS12KeyStore.java

```diff
@@ -1306,7 +1326,7 @@ public KeyStore.Entry engineGetEntry(String alias,
 
         Entry entry = entries.get(alias.toLowerCase(Locale.ENGLISH));
         if (protParam == null) {
-            if (engineIsCertificateEntry(alias)) {
+            if (internalEngineIsCertificateEntry(entry)) {
                 if (entry instanceof CertEntry &&
                     ((CertEntry) entry).trustedKeyUsage != null) {
```

Comment:
`internalEngineIsCertificateEntry` performs exactly the same checks; you can remove these lines.

Reply:
Removed.